main.nf

```diff
@@ -7,96 +7,133 @@ nextflow.enable.dsl=2
+    def loadEnvFile = { envFile ->
+        if (file(envFile).exists()) {
+            file(envFile).readLines().each { line ->
+                if (line && !line.startsWith('#')) {
+                    def parts = line.split('=')
+                    if (parts.size() == 2) {
+                        System.setProperty(parts[0].trim(), parts[1].trim())
+                    }
+                }
+            }
         }
-        repoCh = Channel.value(params.repo_url)
-    } else {
-        throw new IllegalArgumentException("Provide either a sample_sheet or repo_url parameter")
     }
-    
-    // Map each repository URL to a tuple: (repo_url, repo_name, out_dir)
-    def repoTuples = repoCh.map { repo_url ->
-         def repo_name = repo_url.tokenize('/')[-1].replace('.git','')
-         def out_dir = "${params.output_dir}/${repo_name}"
-         tuple(repo_url, repo_name, out_dir)
+
+    // Load .env file
+    loadEnvFile('.env')
+
+    // Parameter validation
+    if (!params.repo_url && !params.sample_sheet) {
+        throw new IllegalArgumentException("ERROR: Provide either a sample_sheet or repo_url parameter")
+    }
+
+    if (params.upload_to_synapse && !params.synapse_folder_id) {
+        throw new IllegalArgumentException("ERROR: synapse_folder_id must be provided when --upload_to_synapse is true.")
+    }
+
+    if (!params.synapse_agent_id) {
+        throw new IllegalArgumentException("ERROR: synapse_agent_id must be provided.")
+    }
+
+    // Validate repository URL format
+    def validateRepoUrl = { url ->
+        if (!url) return false
+        def validUrlPattern = ~/^https:\/\/github\.com\/[^\/]+\/[^\/]+\.git$/
+        return url ==~ validUrlPattern
+    }
+
+    // Extract repository name from URL
+    def getRepoName = { url ->
+        def urlStr = url instanceof List ? url[0] : url
+        return urlStr.tokenize('/')[-1].replace('.git','')
     }
-    
-    // Process each repository with ProcessRepo (clones repo and performs initial checks)
-    def repoOutputs = repoTuples | ProcessRepo
-    
-    // Run the Almanack analysis on each repository
-    def almanackOutputs = repoOutputs | RunAlmanack
-
-    // Collect all unique status files into one list
-    almanackOutputs
-        .map { repo_url, repo_name, out_dir, status_file -> status_file }
-        .collect()
-        .set { allStatusFiles }
-    
-    // Generate the consolidated report from all status files
-    allStatusFiles | GenerateReport
+
+    // Create a channel of repo URLs
+    Channel.from(
+        params.sample_sheet ?
+            file(params.sample_sheet).readLines().drop(1).collect { it.trim() }.findAll { it } :
+            [params.repo_url]
+    ).set { repo_urls }
+
+    // Validate and process each repo
+    repo_urls.map { repo_url ->
+        if (!validateRepoUrl(repo_url)) {
+            throw new IllegalArgumentException("ERROR: Invalid repository URL format. Expected: https://github.com/username/repo.git")
```

Comment:
This doesn't tell the user which repo_url was invalid right? So if you passed a manifest with many urls you'd have to check all of them